validate.sh

```diff
@@ -1,545 +1,3 @@
-print_header "cabal-install: tests"
-
-# this are sorted in asc time used, quicker tests first.
-CMD="$($CABALLISTBIN cabal-install:test:long-tests) $TESTSUITEJOBS --hide-successes"
-(cd cabal-install && timed $CMD) || exit 1
-
-# This doesn't work in parallel either
-CMD="$($CABALLISTBIN cabal-install:test:unit-tests) -j1 --hide-successes"
-(cd cabal-install && timed $CMD) || exit 1
-
-# Only single job, otherwise we fail with "Heap exhausted"
-CMD="$($CABALLISTBIN cabal-install:test:mem-use-tests) -j1 --hide-successes"
-(cd cabal-install && timed $CMD) || exit 1
-
-# This test-suite doesn't like concurrency
-CMD="$($CABALLISTBIN cabal-install:test:integration-tests2) -j1 --hide-successes --with-ghc=$HC"
-(cd cabal-install && timed $CMD) || exit 1
-}
-
-# cabal-install cabal-testsuite
-#######################################################################
-
-step_cli_suite() {
-print_header "cabal-install: cabal-testsuite"
-
-CMD="$($CABALLISTBIN cabal-testsuite:exe:cabal-tests) --builddir=$CABAL_TESTSUITE_BDIR --with-cabal=$($CABALLISTBIN cabal-install:exe:cabal) $TESTSUITEJOBS  --with-ghc=$HC --hide-successes --intree-cabal-lib=$PWD --test-tmp=$PWD/testdb $RTSOPTS"
-(cd cabal-testsuite && timed $CMD) || exit 1
-}
-
-# solver-benchmarks
-#######################################################################
-
-step_solver_benchmarks_tests() {
-print_header "solver-benchmarks: test"
-
-CMD="$($CABALLISTBIN solver-benchmarks:test:unit-tests)"
-(cd Cabal && timed $CMD) || exit 1
-}
-
-step_solver_benchmarks_run() {
-print_header "solver-benchmarks: run"
-
-SOLVEPKG=Chart-diagrams
-CMD="$($CABALLISTBIN solver-benchmarks:exe:hackage-benchmark) --cabal1=$CABAL --cabal2=$($CABALLISTBIN cabal-install:exe:cabal) --trials=5 --packages=$SOLVEPKG --print-trials"
-(cd Cabal && timed $CMD) || exit 1
-}
-
-# Steps dispatcher
-#######################################################################
-
-for step in $STEPS; do
-    case $step in
-        print-config)             step_print_config            ;;
-        print-tool-versions)      step_print_tool_versions     ;;
-        build)                    step_build                   ;;
-        doctest)                  step_doctest                 ;;
-        lib-tests)                step_lib_tests               ;;
-        cli-tests)                step_cli_tests               ;;
-        lib-suite)                step_lib_suite               ;;
-        lib-suite-extras)         step_lib_suite_extras        ;;
-        cli-suite)                step_cli_suite               ;;
-        solver-benchmarks-tests)  step_solver_benchmarks_tests ;;
-        solver-benchmarks-run)    step_solver_benchmarks_run   ;;
-        time-summary)             step_time_summary            ;;
-        *)
-            echo "Invalid step $step"
-            exit 1
-            ;;
-    esac
-done
-
-#######################################################################
+cabal run cabal-validate:exe:cabal-validate -- "$@"
```

Comment:
Do we need to force `bash` for a one-liner? The only change needed is to replace `"$@"` with `${1+"$@"}`.